### PR TITLE
Configure and use uwsgi instead of runserver.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This unfortunately needs to be merged and tested together with https://github.com/mozilla/addons-nginx/pull/14

Fixes https://github.com/mozilla/addons-server/issues/11539
Fixes https://github.com/mozilla/addons-server/issues/6668

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
CircleCI tests are failing because this requires the nginx PR to be merged too :-/

File: docker/uwsgi.ini [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# mysite_uwsgi.ini file
+[uwsgi]
+base            = /code
+chdir           = %(base)
+module          = olympia.wsgi
```
Comment (bqbn) [inline docker/uwsgi.ini]: Production uses `module = olympia.wsgi:application`. Maybe it'll work in your local environment even without `:application`?

Comment (EnTeQuAk) [reply docker/uwsgi.ini]: Yeah, uwsgi defaults to `:application` afaik but I'll add it for explicitness too. Thanks for pointing it out.

**Comment by EnTeQuAk** [comment]
> Production has the following, it may or may not be needed in your local env.

I don't think we're using any uwsgi related plugins, or, rather the default paths and things seem to be working fine :man_shrugging:

**Comment by EnTeQuAk** [comment]
@eviljeff I updated the docker-compose config to link directly to the uwsgi nginx image so that a `docker-compose pull` should give you now the updated image.

Please double-check with `docker-compose exec nginx bash` and `cat /etc/nginx/conf.d/addons.conf | grep uwsgi` and make sure you see something.

File: docker-compose.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ services:
     command: supervisord -n -c /code/docker/supervisor.conf
 
   nginx:
-    image: addons/addons-nginx
+    image: addons/addons-nginx:uwsgi
```
Comment (eviljeff) [inline docker-compose.yml]: reminder to revert this line when the nginx pull is merged.

**Comment by jrbenny35** [comment]
How did you guys get this to work locally? I can not get the site to load at all. Just returns a 502

**Comment by jrbenny35** [comment]
Looks like the docker build hasn't happened yet.

**Comment by EnTeQuAk** [comment]
> Looks like the docker build hasn't happened yet.

Thanks for flagging this, I have indeed to build this manually… completely forgot this. The autobuilds are still broken for addons-nginx.

**Comment by EnTeQuAk** [comment]
```
[git:master] ~/Projects/addons-nginx> docker build -t addons/addons-nginx:latest .
Sending build context to Docker daemon  197.6kB
Step 1/3 : FROM nginx
latest: Pulling from library/nginx
0a4690c5d889: Pull complete
9719afee3eb7: Pull complete
44446b456159: Pull complete
Digest: sha256:b4b9b3eee194703fc2fa8afa5b7510c77ae70cfba567af1376a573a967c03dbb
Status: Downloaded newer image for nginx:latest
 ---> 98ebf73aba75
Step 2/3 : EXPOSE 80
 ---> Running in e6b3343729fe
Removing intermediate container e6b3343729fe
 ---> 813669237118
Step 3/3 : ADD addons.conf /etc/nginx/conf.d/addons.conf
 ---> 7dc11777d896
Successfully built 7dc11777d896
Successfully tagged addons/addons-nginx:latest
[git:master] ~/Projects/addons-nginx> docker push addons/addons-nginx:latest
The push refers to repository [docker.io/addons/addons-nginx]
41871ae6cdee: Pushed
589561a3ffb4: Layer already exists
ef7dbb0cfc81: Layer already exists
d56055da3352: Layer already exists
latest: digest: sha256:08fd4a91637d00c20a0f119739a4083c76e3b0abd716c3c090d8575e0803315f size: 1156
```

The nginx image is now up-to-date. Please re-pull! Sorry about that.

**Pull request merged**